### PR TITLE
Add support for Redis Cache

### PR DESCRIPTION
**Pull request by alexandre-vasc**
This PR closes dojot/dojot#69

### CODE REVIEW

File: auth/conf.py [hunk 1]
```diff
@@ -5,15 +5,28 @@
 
 
 # database related configuration
-dbName = os.environ.get("DB_NAME", "postgres")
-dbUser = os.environ.get("DB_USER", "auth")
-dbPdw = os.environ.get("DB_PWD", "")
-dbHost = os.environ.get("DB_HOST", "postgres")
+dbName = os.environ.get("AUTH_DB_NAME", "postgres")
```
Comment (alexandre-vasc) [inline auth/conf.py]: Now that Auth is growing,  there is a lot of configuration available. 
I think its is a bit messy to configure all these stuff by environments variables. Maybe its time to create a configuration file?

File: auth/controller/PDPController.py [hunk 2]
```diff
@@ -22,55 +23,62 @@ def checkRequest(pdpRequest):
 
 def pdpMain(dbSession, pdpRequest):
     checkRequest(pdpRequest)
-    try:
```
Comment (alexandre-vasc) [inline auth/controller/PDPController.py]: Moved JWT checks to AuthenticationController

File: auth/database/Cache.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    REDIS_URL = ('redis://' + conf.cacheUser + ':' + conf.cachePdw
+                 + '@' + conf.cacheHost + ':6379/' + conf.cacheDatabase)
+    app.config['DBA_URL'] = REDIS_URL
+    redis_store = FlaskRedis(app, config_prefix='DBA', strict=False,
+                             encoding="utf-8", socket_keepalive=True,
+                             charset="utf-8", decode_responses=True)
+
+
+elif (conf.cacheName == 'NOCACHE'):
+    print("Warning. Cache policy set to NOCACHE."
```
Comment (alexandre-vasc) [inline auth/database/Cache.py]: Its OK to use prints on those warnings? or should we use a logger?

File: auth/conf.py (hunk 1 above)
Comment (mmagr) [reply auth/conf.py]: Yes, having that option sounds like a good idea.
However, I'd advocate to keeping things configurable through environment variables as well (seems to be a kind of 'de facto' standard for docker-deployed services.

File: auth/database/Cache.py (hunk 3 above)
Comment (mmagr) [reply auth/database/Cache.py]: Using the logger should be better (depending on the infrastructure, prints will not be "printed" until the master process crashes, making runtime management a tad hard to do.

**Pull request merged**